src/HOCON/Impl/Utils.cs

```diff
@@ -124,6 +124,17 @@ public static bool Contains(this string s, char c)
             return s.IndexOf(c) != -1;
         }
 
+        public static bool ContainsHoconWhitespaceExceptNewLine(this string s)
```

Comment:
This is is covered by the NeedTripleQuotes() string extension.

Reply:
Again, I was worried that a key may contain whitespace (such as `my key`). `NeedTripleQuotes` calls `NeedQuotes`, which I understsand does not check for whitespace. Did not cross my mind, but would a path such as "foo.my key.bar` be valid (or does it have to be `foo."my key".bar`)?

Reply:
foo."my.key".bar and foo."my long spacious key".bar are both legal, but not foo."""my\nkey""".bar. Your concern is correct, there is a bug in Path parsing. My point is that, instead of making another extension method, you should fix the NeedTripleQuotes() method instead